### DISCUSSION TITLE
steam安装不上，主站的arm游戏怎么才能打开

### DISCUSSION THREAD

**Discussion by N00k0428** [Gamuxorg/bbs#418] [建议和闲聊]
缺失steam_run啥的运行库，

**Comment by hyh19962008** [comment]
哪个游戏

Best
________________________________
From: ltjian-lttj ***@***.***>
Sent: Tuesday, July 25, 2023 10:22:47 PM
To: Gamuxorg/bbs ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [Gamuxorg/bbs] steam安装不上，主站的arm游戏怎么才能打开 (Discussion #418)


缺失steam_run啥的运行库，

―

**Reply by N00k0428** [reply]
想玩arm列表的gta3，这个标注了steam。
尝试了在麒麟v10_x86虚拟机运行主站的纪念碑谷2，缺乏steam的运行库。因为没能成功安装steam，steam_latest.deb下了很多依赖，但最后还是没能启动steam。
工作仓库是arrch64的麒麟，没网络，想安装steam估计更困难

**Reply by kero990** [reply]
> 想玩arm列表的gta3，这个标注了steam。 尝试了在麒麟v10_x86虚拟机运行主站的纪念碑谷2，缺乏steam的运行库。因为没能成功安装steam，steam_latest.deb下了很多依赖，但最后还是没能启动steam。 工作仓库是arrch64的麒麟，没网络，想安装steam估计更困难

gta3是当初一帮大佬开源复刻的，还被r星告了，这个是不需要steam的
而且arm linux现在也没法运行steam，你装上也没用，因为steam上的游戏就算有linux，也是x86的，不可能用steam直接启动
至于steam-runtime，是个开源项目，用来跑steam游戏的环境，只要有这个就行

**Comment by hyh19962008** [comment]
可以命令行运行一下提供报错信息吗

Best
________________________________

**Reply by N00k0428** [reply]
Steam needs to install these additional packages:
libc6:amd64 libc:i386 libeg11:amd64……
按回车继续安装后，最终提示：
libg14-mesg-dri:i386: 依赖: libllvm12:i386(>=1:9~svn298832-1~),但是它将不会被安装

**Reply by N00k0428** [reply]
今天还把主站arm架构的盐与避难所（没标注steam）拿去工作电脑（arrch64）在命令行运行了了一下：
找不到
/lib/aarch64-linux-gun/libm.so.6: version 'GLIBC_2.29'
/lib/aarch64-linux-gun/libm.so.6: version 'GLIBC_2.27'
/lib/aarch64-linux-gun/libc.so: version 'GLIBC_2.27'
/lib/aarch64-linux-gun/libc.so: version 'GLIBC_2.28'

**Reply by shouhuanxiaoji** [reply]
系统信息是否可以发一下
lscpu
apt-cache info libc6
看起来是libc比较老，而盐与避难所要求的高一些
这个问题能否解决，要看对于glibc的版本要求，是模拟器发起还是游戏本身需求。

**Comment by hyh19962008** [comment]
盐与避难所是你的libc版本太老了，它要求libc 2.29以上的，这个只能升级系统解决

Best
________________________________

**Comment by Sun-is-rising** [comment]
可以尝试安装flatpak版steam，先安装flatpak:
`sudo apt install flatpak`
再换源：
`sudo flatpak remote-modify flathub --url=https://mirror.sjtu.edu.cn/flathub`
然后安装steam:
`flatpak install com.valvesoftware.Steam`
flatpak版应用不会出现依赖问题，因为所有的包都会安装在容器中，但是软件体积较大。
也有可能是显卡驱动问题，今天我安装了原生英伟达驱动，原版steam可以启动了。
